### DISCUSSION TITLE
Confirm that 4.x won't require tzlocal

### DISCUSSION THREAD

**Discussion by davetapley** [agronholm/apscheduler#787] [Q&A]
Can someone confirm that upcoming 4.x won't require `tzlocal` any more?

It's tangentially mentioned in the `v3.x to v4.0` docs here:
https://github.com/agronholm/apscheduler/blob/77a80020184fbc436db98800aa992b6b3a98a53c/docs/migration.rst?plain=1#L89-L93

---

I ask because TIL `3.x` requires `tzlocal`, which requires `tzdata`, which comprises 603 files.

That is nearly half the files in my app when built with PyInstaller, doubling the time it takes to extract, despite my app being Python 3.11 and so already having the `zoneinfo` builtin with all the zones  🤪 

I was going to look for a way to mitigate including all of `tzdata`'s files, but if 4.x removes the dependency then I'll just wait 🤞🏻

**Comment by agronholm** [comment]
How would APScheduler get the local timezone then?

**Reply by davetapley** [reply]
I think `datetime.astimezone(None)`?

> If called without arguments (or with `tz=None`) the system local timezone is assumed for the target timezone. 

https://docs.python.org/3.11/library/datetime.html#datetime.datetime.astimezone

---

To be sure: I have no knowledge of internals of APScheduler (or why it needed `tzlocal` in the first place) 😁

**Reply by agronholm** [reply]
Unfortunately that uses a `datetime.timezone` object which is just a fixed UTC offset, so daylight saving time calculations won't work right.
